junit5/src/test/java/org/jboss/weld/junit5/auto/AutoConfigWithWeldSetupTest.java

```diff
@@ -0,0 +1,63 @@
+package org.jboss.weld.junit5.auto;
+
+import java.lang.reflect.Field;
+import java.util.Optional;
+
+import org.jboss.weld.junit5.WeldInitiator;
+import org.jboss.weld.junit5.WeldSetup;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.AfterEachCallback;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.platform.engine.support.hierarchical.ThrowableCollector;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertInstanceOf;
+import static org.junit.jupiter.api.Assertions.fail;
+
+/**
+ * NOTE - this test is one massive hack because JUnit, at the time of writing, doesn't allow to assert that an
+ * exception was throws inside an extension. The test is fragile and can break because of internal JUnit 5 changes
+ * in which case we might want to revisit if we want to keep it or whether there is a better way.
+ *
+ * @author bkautler
+ */
+@EnableAutoWeld
+class AutoConfigWithWeldSetupTest {
```

Comment:
@Vampire my apologies, I've set you to an impossible task and I didn't even realize it.
It seems you cannot assert that an exception is thrown from inside an extension. Apparently, that didn't stop you and I have to say you handled it really well :D

Since you took the work, let's keep it. I've just pushed a small note to the top of the test so that we aren't as surprised once it starts failing in some newer junit version.

Reply:
Don't worry, I like to dig in the mud :-D